# Graded assignment 1

Sjoerd Geraads, 2008618


# Packages

We need the package `tidyverse`

In [3]:
library(tidyverse) 






# Assignment 1

First, we generate some random data:

In [4]:

df1 <- as.data.frame(matrix(runif(20*2), ncol=20) )
head(df1)

V1        V2        V3        V4         V5         V6        V7       
1 0.9985231 0.8662440 0.6447793 0.08510927 0.02888499 0.7105097 0.5777318
2 0.7933338 0.5539696 0.5919835 0.57506257 0.63489761 0.2196523 0.4454652
  V8        V9        V10       V11        V12       V13       V14      
1 0.4847134 0.9773228 0.4471305 0.86157256 0.8933256 0.8109158 0.8110618
2 0.6680473 0.7160994 0.9049130 0.06581781 0.2319006 0.2136249 0.6354761
  V15       V16       V17       V18       V19       V20      
1 0.6337535 0.2354914 0.6244068 0.9971074 0.6769139 0.2405547
2 0.9663184 0.7570708 0.9441425 0.4638785 0.9386224 0.9290426


_df1_ is a wide dataframe. The goal of the assignment is to change _df1_ into a long dataframe _df2_. And show the first rows of the dataframe using `head(df2)`

The dataframe _df2_ should look like this:

| Key | Value      |
|-----|------------|
| V1  | 0.09413864 |
| V1  | 0.69782526 |
| V2  | 0.23460454 |
| V2  | 0.51520514 |
| V3  | 0.63181844 |
| V3  | 0.22807211 |

In [7]:
df2 <- df1 %>%
  gather(Key, Value, V1:V20)

head(df2)

Key Value    
1 V1  0.9985231
2 V1  0.7933338
3 V2  0.8662440
4 V2  0.5539696
5 V3  0.6447793
6 V3  0.5919835

## Assignment 2

We will generate some data:

In [8]:
df3 <- as.data.frame(read.table(header=TRUE, text='
      id        V_1       V_2                
          1  42.467573  -87.810047     
          2  42.049148  -88.273029
          3  39.110539  -90.324080
                       '))

head(df3)

id V_1      V_2      
1 1  42.46757 -87.81005
2 2  42.04915 -88.27303
3 3  39.11054 -90.32408

In [10]:
df4 <- as.data.frame(read.table(header=TRUE, text='
     V_3            V_4         id                 
      one         1h            1
      two         1h            1
      three       8h            2
      four        1h            4
'))

head(df4)

V_3   V_4 id
1 one   1h  1 
2 two   1h  1 
3 three 8h  2 
4 four  1h  4


You will have to merge these files, returning all rows and all columns from both _df3_ and _df5_. Where there are not matching values, return _NA_ for the missings. 

This should be the end result:

| id | V_1      | V_2       | V_3   | V_4 |
|----|----------|-----------|-------|-----|
| 1  | 42.46757 | -87.81005 | one   | 1h  |
| 1  | 42.46757 | -87.81005 | two   | 1h  |
| 2  | 42.04915 | -88.27303 | three | 8h  |
| 3  | 39.11054 | -90.32408 | NA    | NA  |
| 4  | NA       | NA        | four  | 1h  |



In [11]:
df5 <- full_join(df3, df4)
df5

Joining, by = "id"


id V_1      V_2       V_3   V_4
1 1  42.46757 -87.81005 one   1h 
2 1  42.46757 -87.81005 two   1h 
3 2  42.04915 -88.27303 three 8h 
4 3  39.11054 -90.32408 NA    NA 
5 4        NA        NA four  1h

End of notebook